src/amdgpu.jl

```diff
@@ -0,0 +1,106 @@
+# implementation of the GPUCompiler interfaces for generating GCN code
+
+## target
+
+export AMDGPUCompilerTarget
+
+Base.@kwdef struct AMDGPUCompilerTarget <: AbstractCompilerTarget
+    dev_isa::String
+
+    runtime_module::Base.Module
+
+    # hooks for exposing functionality to AMDGPUnative.jl
+    # FIXME: this isn't particularly nice, duplicating fields and functions
+    #        from the abstract interface over this job to AMDGPUnative...
+    isintrinsic_hook::Union{Nothing,Function}=nothing
+end
+
+AMDGPUCompilerTarget(dev_isa; kwargs...) =
+    AMDGPUCompilerTarget(; dev_isa=dev_isa, kwargs...)
+
+llvm_triple(::AMDGPUCompilerTarget) = "amdgcn-amd-amdhsa"
+
+function llvm_machine(target::AMDGPUCompilerTarget)
+    triple = llvm_triple(target)
+    t = Target(triple)
+
+    cpu = target.dev_isa
+    feat = ""
+    optlevel = LLVM.API.LLVMCodeGenLevelDefault
+    reloc = LLVM.API.LLVMRelocPIC
+    tm = TargetMachine(t, triple, cpu, feat, optlevel, reloc)
+    asm_verbosity!(tm, true)
+
+    return tm
+end
+
+llvm_datalayout(::AMDGPUCompilerTarget) =
```

Comment:
You only need to override the datalayout if the one selected by LLVM isn't the correct one -- chances are it is.